test/dlc_tests/sanity/test_remote_function_compatibility.py

```diff
@@ -0,0 +1,40 @@
+from packaging.version import Version
+from packaging.specifiers import SpecifierSet
+import pytest
+
+from invoke import run
+from test.test_utils import get_python_version_from_image_uri
+
+
+@pytest.mark.usefixtures("sagemaker_only")
+@pytest.mark.model("N/A")
+@pytest.mark.integration("remote_function")
+def test_remote_function(image):
+    """
+    Test to check compatibility of sagemaker training images with sagemaker remote function.
+    """
+    python_version = get_python_version_from_image_uri(image).replace("py", "")
+    python_version = int(python_version)
+
+    if python_version < 37 or "training" not in image:
```

Comment:
Instead of checking for `"training" not in image`, please use `def test_remote_function(training):` instead, which ensures that only training images are passed to the function.

Reply:
Updated